Makefile

```diff
@@ -73,50 +64,71 @@ gitstat:
 	@git status
 
 vendor:
-	$(BUILD_CMD) go mod download
+ifeq ($(BUILD),local)
+	$(GO) go mod download
+endif
 
 build: vendor $(BINARY)
 
 $(BINDIR):
 	mkdir -p $(BINDIR)
 
 $(BINARY): $(BINDIR)
-	$(BUILD_CMD) go build -v -i -o $(BINARY)
+ifneq ($(BUILD),local)
+	$(MAKE) image
+	# because there is no way to `docker extract` or `docker cp` from an image
+	CID=$$(docker create $(IMGTAG)) && \
+	docker cp $${CID}:/aws-asg-roller $(BINARY) && \
```

Comment:
> I don't follow you. What do you mean by "building the go binary more than once"?

Specifically referring to the approach here and related comments https://github.com/deitch/aws-asg-roller/pull/4#issuecomment-488213555
Namely:
>  If it is not (only) an image, e.g. you want the binary locally, you have to do extra work to get there.

The suggestion I'm trying to make is not to look at using a multistage build as a mean of redefining the way binaries are built wholesale, but solely as a way of making your docker build process self contained.

Practically speaking that looks something like dropping the `ifneq ($(BUILD),local)` block here and sticking with the `go build` which you can paramterize for the os,arch etc.

Reply:
In large meaning there shouldn't be any significant (if any) changes to the makefile required to adopt multistage builds with for your docker images. That was at least the intent with #4